File: baekjoon/python/13414.py
import sys; input = sys.stdin.readline

K, L = map(int, input().split())

waiting = {}
for i in range(L): 
    student = input().rstrip()
    if student in waiting:
        waiting.pop(student)
        waiting[student] = i
    else: waiting[student] = i

result = sorted(list(waiting.items()), key=lambda x: x[1])
for student in result[:K]: print(student[0])

File: baekjoon/javascript/13414.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  let [K, L] = input
    .shift()
    .split(" ")
    .map(x => +x);
  const waiting = new Map();
  for (let i in input) waiting.set(input[i], +i);
  result = Array.from(waiting.entries()).sort((a, b) => a[1] - b[1]);
  console.log(
    result
      .slice(0, K)
      .map(x => x[0])
      .join("\n"),
  );
});
